cesium/internal/domain/file_controller.go

```diff
@@ -50,16 +51,45 @@ func openFileController(cfg Config) (*fileController, error) {
 	if err != nil {
 		return nil, err
 	}
+	counter, err := xio.NewInt32Counter(counterF)
+	if err != nil {
+		return nil, err
+	}
 	fc := &fileController{
 		Config:      cfg,
-		counter:     xio.NewInt32Counter(counterF),
+		counter:     counter,
 		counterFile: counterF,
 	}
 	fc.writers.open = make(map[uint16]controlledWriter, cfg.MaxDescriptors)
 	fc.writers.release = make(chan struct{}, cfg.MaxDescriptors)
 	fc.readers.open = make(map[uint16][]controlledReader)
 	fc.readers.release = make(chan struct{}, cfg.MaxDescriptors)
-	return fc, nil
+
+	fc.writers.unopened, err = fc.scanUnopenedFiles()
+	return fc, err
+}
+
+func (fc *fileController) scanUnopenedFiles() (map[uint16]struct{}, error) {
+	unopened := make(map[uint16]struct{})
+	for i := 1; i <= int(fc.counter.Value()); i++ {
+		e, err := fc.Config.FS.Exists(fileKeyToName(uint16(i)))
+		if err != nil {
+			return unopened, err
+		}
+		if !e {
+			continue
+		}
+
+		s, err := fc.Config.FS.Stat(fileKeyToName(uint16(i)))
+		if err != nil {
+			return unopened, err
+		}
+		if s.Size() < int64(fc.FileSize) {
```

Comment:
OK, yes, agreed, but I also don't like the idea of having two fields to control the file size (both limit AND a proportion of the limit as a threshold) – so for now I'm going to hardcode a value of 0.9 for all such occurences: since the defauly value is 1gb, 900mb seems like a reasonable size to not allow new writers.